libs/native/src/main21/java/org/elasticsearch/nativeaccess/jdk/JdkVectorLibrary.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.nativeaccess.jdk;
+
+import org.elasticsearch.nativeaccess.lib.VectorLibrary;
+
+import java.lang.foreign.FunctionDescriptor;
+import java.lang.foreign.MemorySegment;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+import static java.lang.foreign.ValueLayout.ADDRESS;
+import static java.lang.foreign.ValueLayout.JAVA_BYTE;
+import static java.lang.foreign.ValueLayout.JAVA_INT;
+import static org.elasticsearch.nativeaccess.jdk.LinkerHelper.downcallHandle;
+
+public final class JdkVectorLibrary implements VectorLibrary {
+
+    static {
+        loadLibrary();
+    }
+
+    @SuppressWarnings("removal")
+    private static void loadLibrary() {
+        AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
+            System.loadLibrary("vec");
+            return null;
+        });
+    }
+
+    public JdkVectorLibrary() {}
+
+    private static final MethodHandle stride$mh = downcallHandle("stride", FunctionDescriptor.of(JAVA_INT));
+    private static final MethodHandle dot8s$mh = downcallHandle("dot8s", FunctionDescriptor.of(JAVA_INT, ADDRESS, ADDRESS, JAVA_INT));
+
+    // Stride of the native implementation - consumes this number of bytes per loop invocation.
+    // There must be at least this number of bytes/elements available when going native
+    static final int STRIDE = 32;
```

Comment:
Can't we initialize it with `stride()` instead of making this a constant and asserting?
This way different architectures could use different strides.

Reply:
I preferred to use the constant value, since the compiler can then constant-fold it, but that's not a super important optimisation. I can look at removing it.